ynr/apps/people/merging.py

```diff
@@ -0,0 +1,202 @@
+import json
+
+from django.conf import settings
+from django.contrib.admin.utils import NestedObjects
+from django.db import IntegrityError, connection, transaction
+
+from candidates.models import (
+    LoggedAction,
+    PersonRedirect,
+    UnsafeToDelete,
+    merge_popit_people,
+)
+from candidates.models.versions import get_person_as_version_data
+from candidates.views.version_data import get_change_metadata, get_client_ip
+
+
+class InvalidMergeError(ValueError):
+    """
+    Raised when merging two people would cause invalid data or some other
+    oddness
+    """
+
+    pass
+
+
+class PersonMerger:
+    """
+    Deals with merging two people, ensuring that no data is lost
```

Comment:
In this comment can we also specify that we'll merge the properties of `source_person` into the `dest_person` object and delete `source_person`